src/Main.elm

```diff
@@ -1,10 +1,362 @@
 module Main exposing (main)
```

Comment:
Hi, thx for finishing the challenge. Its a good start. Below you can find hints of how code could be improved.

* organize file around business logic / data structure -eg Auidence.elm should contain decoders and Cmd related to it https://youtu.be/XpDsk374LDE?t=443
* using elm-css https://package.elm-lang.org/packages/rtfeldman/elm-css/latest/Css instead of separate css file
* its prefered to have naming of Msgs in form of SomethingHappend not DoSomething https://www.youtube.com/watch?v=w6OVDBqergc&ab_channel=ElmintheSpring
*  instead of having errorMessage field - use RemoteData https://package.elm-lang.org/packages/krisajenkins/remotedata/latest/
*  GoBack - could contain the information where to go to (we have to have that information available in View because we are rendering AudienceFolder which contains that information)
*  view
   * functions should not accept the whole model, just parts which are necessary for the function to work => easier to debug in future
   * could be organized/simplified
       * no need to split into such a small functions (its not javascript :))
       * no need to have separate function for different situations, its enough to have one view only with filters changing based on situation. Have a look how condensed the view can be
```elm
view : Model -> Html Msg
view model =
    div [ class "table" ]
        (List.concat
            [ [ div []
                    [ button [ class "category-button", onClick (SelectCategory Audience.Authored) ] [ text "Authored" ]
                    , button [ class "category-button", onClick (SelectCategory Audience.Shared) ] [ text "Shared" ]
                    , button [ class "category-button", onClick (SelectCategory Audience.Curated) ] [ text "Curated" ]
                    ]
              ]
            , if model.currentFolderId /= Nothing then
                [ div
                    [ class "goBack-button", onClick GoBack ]
                    [ text "Go Up" ]
                ]

              else
                []
            , [ div []
                    ((model.audienceFolder
                        |> List.filter (\folder -> folder.parent == model.currentFolderId)
                        |> List.filter (\_ -> model.selectedCategory /= Audience.Shared)
                        |> List.map
                            (\folder ->
                                button
                                    [ class "folder-button"
                                    , onClick (OpenFolder folder.id)
                                    ]
                                    [ text folder.name ]
                            )
                     )
                        ++ (model.audience
                                |> List.filter (\audience -> audience.folder == model.currentFolderId && model.selectedCategory == audience.type_)
                                |> List.map
                                    (\audience ->
                                        button
                                            [ class "audience-button"
                                            ]
                                            [ text audience.name ]
                                    )
                           )
                    )
              ]
            ]
        )
```

Reply:
Hello, thank you for your advices, I will bear in mind these tips 😄